aten/src/ATen/native/cuda/GridSampler.cu

```diff
@@ -743,9 +836,15 @@ namespace {
         }
 
         // un-normalize grad_grid values back to [-1, 1] constraints
-        gix = gix * (inp_W - 1) / 2;
-        giy = giy * (inp_H - 1) / 2;
-        giz = giz * (inp_D - 1) / 2;
+        if (align_corners) {
```

Comment:
I feel it's good to add another helper function to handle this logic in one place, I've seen it in a few files here.